splainer-search.js

```diff
@@ -1037,6 +1038,62 @@ angular.module('o19s.splainer-search')
 
 'use strict';
 
+angular.module('o19s.splainer-search')
+  .service('queryTemplateSvc', function queryTemplateSvc() {
+    var self      = this;
+    self.hydrate = hydrate;
+
+    var defaultConfig = {
+      encodeURI: false,
+      defaultKw: '""',
+    };
+
+    function encode(queryPart, config) {
+      if (config.encodeURI) {
+        return encodeURIComponent(queryPart);
+      } else {
+        return queryPart;
+      }
+    }
+
+
+    function keywordMapping(queryText) {
+      var queryTerms    = queryText.split(/[ ,]+/);
+      var maxKeywords = 10;
+      var numTerms = queryTerms.length;
+      for (var i = numTerms; i < maxKeywords; i++) {
+        queryTerms.push(null);
+      }
+      return queryTerms;
+    }
+
+    function hydrate(template, queryText, config) {
+      if (!config) {
+        config = defaultConfig;
+      }
+
+      if (queryText === null || angular.isUndefined(queryText)) {
+        return template;
+      }
+
+      var replaced  = template.replace(/#\$query##/g, encode(queryText, config));
+      var idx = 0;
+      angular.forEach(keywordMapping(queryText), function(queryTerm) {
```

Comment:
can we flip the logic behind the loop?
Determine how many `/#\$query\d+##/` match, that gives you the number `n` of queries that need to be replaced.
Loop `n` times, replace `var regex = new RegExp('#\\$query' + (i) + '##', 'g');` with `terms[i]`
This way you won't have to set `var maxKeywords = 10;` and you won't have to fill the array with `null`s
Less looping == faster code.